pkg/auth/service_test.go

```diff
@@ -756,3 +760,226 @@ func benchmarkListEffectivePolicies(b *testing.B, s *auth.DBAuthService, userNam
+					Id:   0,
+					Name: u,
+				}
+			}
+			returnedUserList := &auth.UserList{
+				Pagination: auth.Pagination{},
+				Results:    returnedUsers,
+			}
+			mockClient.EXPECT().ListUsersWithResponse(gomock.Any(), gomock.Any()).Return(&auth.ListUsersResponse{
+				Body: nil,
+				HTTPResponse: &http.Response{
+					StatusCode: tt.responseStatusCode,
+				},
+				JSON200:     returnedUserList,
+				JSON401:     nil,
+				JSONDefault: nil,
+			}, nil)
+
+			ctx := context.Background()
+			ID := int64(12)
+			gotUser, err := s.GetUserByID(ctx, ID)
+			if !errors.Is(err, tt.expectedErr) {
+				t.Fatalf("GetUserById(%d): expected err: %s got: %s", ID, tt.expectedErr, err)
+			}
+			if err != nil {
+				return
+			}
+
+			expectedUserID := int64(0)
+			if gotUser.ID != expectedUserID {
+				t.Fatalf("expected user id:%d, got:%d", expectedUserID, gotUser.ID)
+			}
+		})
+	}
+
+}
+
+func TestAPIAuthService_GetUserByEmail(t *testing.T) {
+	mockClient, s := GetApiService(t)
+
+	tests := []struct {
+		name               string
+		responseStatusCode int
+		users              []string
+		expectedUserID     int64
+		expectedErr        error
+	}{
+		{
+			name:               "one user",
+			responseStatusCode: http.StatusOK,
+			users:              []string{"one"},
+			expectedUserID:     0,
+			expectedErr:        nil,
+		}, {
+			name:               "no users",
+			responseStatusCode: http.StatusOK,
+			users:              []string{},
+			expectedUserID:     0,
+			expectedErr:        db.ErrNotFound,
+		},
+		{
+			name:               "two responses",
+			responseStatusCode: http.StatusOK,
+			users:              []string{"one", "two"},
+			expectedUserID:     0,
+			expectedErr:        auth.ErrNonUnique,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			returnedUsers := make([]auth.User, len(tt.users))
+			for i, u := range tt.users {
+				returnedUsers[i] = auth.User{
+					Id:   0,
+					Name: u,
+				}
+			}
+			returnedUserList := &auth.UserList{
+				Pagination: auth.Pagination{},
+				Results:    returnedUsers,
+			}
+			mockClient.EXPECT().ListUsersWithResponse(gomock.Any(), gomock.Any()).Return(&auth.ListUsersResponse{
+				Body: nil,
+				HTTPResponse: &http.Response{
+					StatusCode: tt.responseStatusCode,
+				},
+				JSON200:     returnedUserList,
+				JSON401:     nil,
+				JSONDefault: nil,
+			}, nil)
+
+			ctx := context.Background()
+			ID := int64(12)
+			gotUser, err := s.GetUserByEmail(ctx, "mail@test.com")
+			if !errors.Is(err, tt.expectedErr) {
+				t.Fatalf("GetUserById(%d): expected err: %s got: %s", ID, tt.expectedErr, err)
+			}
+			if err != nil {
+				return
+			}
+
+			expectedUserID := int64(0)
+			if gotUser.ID != expectedUserID {
+				t.Fatalf("expected user id:%d, got:%d", expectedUserID, gotUser.ID)
+			}
+		})
+	}
+
+}
+
+func GetApiService(t *testing.T) (*mock.MockClientWithResponsesInterface, *auth.APIAuthService) {
+	t.Helper()
+	ctrl := gomock.NewController(t)
```

Comment:
In order to verify that the mock expections matched we need to call
```go
defer ctrl.Finish()
```go
So I think you will have to return it so the test can call the above Finish method when the test is done.